File: scripts/check-table-structure.js
const mysql = require('mysql2/promise');

async function checkTableStructure() {
  const connection = await mysql.createConnection({
    host: '127.0.0.1',
    user: 'root',
    password: 'Snqig521223',
    database: 'vnerpdacahng'
  });

  const tables = [
    'crm_customer',
    'pur_request',
    'pur_request_item',
    'inv_warehouse',
    'inv_inventory',
    'inv_inventory_log',
    'prd_standard_card',
    'sys_employee',
    'sample_order'
  ];

  for (const table of tables) {
    try {
      const [columns] = await connection.execute(`DESCRIBE ${table}`);
      console.log(`\n=== ${table} ===`);
      columns.forEach(col => {
        console.log(`  ${col.Field}: ${col.Type} ${col.Null === 'NO' ? 'NOT NULL' : 'NULL'} ${col.Default ? `DEFAULT ${col.Default}` : ''}`);
      });
    } catch (err) {
      console.log(`\n=== ${table} === 错误: ${err.message}`);
    }
  }

  await connection.end();
}

checkTableStructure().catch(console.error);

File: scripts/check_table_structure.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""查询表结构"""
import mysql.connector

conn = mysql.connector.connect(
    host='127.0.0.1',
    port=3306,
    user='root',
    password='Snqig521223',
    database='vnerpdacahng'
)

cursor = conn.cursor()

# 查询 inv_material 表结构
print("inv_material 表结构:")
cursor.execute("DESCRIBE inv_material")
for row in cursor.fetchall():
    print(f"  {row[0]}: {row[1]}")

print("\ninv_material_category 表结构:")
cursor.execute("DESCRIBE inv_material_category")
for row in cursor.fetchall():
    print(f"  {row[0]}: {row[1]}")

print("\ninv_warehouse 表结构:")
cursor.execute("DESCRIBE inv_warehouse")
for row in cursor.fetchall():
    print(f"  {row[0]}: {row[1]}")

conn.close()
